frontend/src/modules/2DViewer/layers/implementations/layers/DrilledWellTrajectoriesLayer/DrilledWellTrajectoriesLayer.ts

```diff
@@ -0,0 +1,125 @@
+    }
+
+    getSettingsContext() {
+        return this._layerDelegate.getSettingsContext();
+    }
+
+    getItemDelegate(): ItemDelegate {
+        return this._itemDelegate;
+    }
+
+    getLayerDelegate(): LayerDelegate<DrilledWellTrajectoriesSettings, WellboreTrajectory_api[]> {
+        return this._layerDelegate;
+    }
+
+    doSettingsChangesRequireDataRefetch(
+        prevSettings: DrilledWellTrajectoriesSettings,
+        newSettings: DrilledWellTrajectoriesSettings
+    ): boolean {
+        return !isEqual(prevSettings, newSettings);
+    }
+
+    makeBoundingBox(): BoundingBox | null {
+        const data = this._layerDelegate.getData();
+        if (!data) {
+            return null;
+        }
+
+        const bbox: BoundingBox = {
+            x: [Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER],
+            y: [Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER],
+            z: [Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER],
+        };
+
+        for (const trajectory of data) {
+            for (const point of trajectory.eastingArr) {
+                bbox.x[0] = Math.min(bbox.x[0], point);
+                bbox.x[1] = Math.max(bbox.x[1], point);
+            }
+            for (const point of trajectory.northingArr) {
+                bbox.y[0] = Math.min(bbox.y[0], point);
+                bbox.y[1] = Math.max(bbox.y[1], point);
+            }
+            for (const point of trajectory.tvdMslArr) {
+                bbox.z[0] = Math.min(bbox.z[0], point);
+                bbox.z[1] = Math.max(bbox.z[1], point);
+            }
+        }
+
+        return bbox;
+    }
+
+    fechData(queryClient: QueryClient): Promise<WellboreTrajectory_api[]> {
+        const workbenchSession = this.getSettingsContext().getDelegate().getLayerManager().getWorkbenchSession();
+        const ensembleSet = workbenchSession.getEnsembleSet();
+        const settings = this.getSettingsContext().getDelegate().getSettings();
+        const ensembleIdent = settings[SettingType.ENSEMBLE].getDelegate().getValue();
+        const selectedWellboreHeaders = settings[SettingType.SMDA_WELLBORE_HEADERS].getDelegate().getValue();
+        let selectedWellboreUuids: string[] = [];
+        if (selectedWellboreHeaders) {
+            selectedWellboreUuids = selectedWellboreHeaders.map((header) => header.wellboreUuid);
+        }
+
+        let fieldIdentifier: string | null = null;
+        if (ensembleIdent) {
+            const ensemble = ensembleSet.findEnsemble(ensembleIdent);
+            if (ensemble) {
+                fieldIdentifier = ensemble.getFieldIdentifier();
+            }
+        }
+
+        const queryKey = ["getWellTrajectories", fieldIdentifier];
+        this._layerDelegate.registerQueryKey(queryKey);
+
+        const promise = queryClient
+            .fetchQuery({
+                queryKey,
+                queryFn: () => apiService.well.getWellTrajectories(fieldIdentifier ?? ""),
+                staleTime: 1800000, // TODO
```

Comment:
TODO what?

Reply:
No idea. @HansKallekleiv can this be removed or would you like to extend the comment?